corehq/apps/userreports/views.py

```diff
@@ -736,6 +737,11 @@ def delete_report(request, domain, report_id):
         ),
         extra_tags='html'
     )
+
+    report_configs = ReportConfig.by_domain_and_owner(
+        domain, request.couch_user.get_id, "configurable")
+    (rc.delete() for rc in report_configs if rc.subreport_slug == config.get_id)
```

Comment:
nit: This might be clearer as a for-loop.  I see what you're doing but there aren't many uses of this type of construction in cchq

Reply:
Would be better to use one of these: https://github.com/dimagi/commcare-hq/blob/194506d4bbf2edde0a81a9e6cc66d5752784a7fa/corehq/ex-submodules/dimagi/utils/couch/database.py#L86-L119

Reply:
Would ```iter_bulk_delete``` execute the code specific to the implementation of ```ReportConfig.delete```?  See: https://github.com/dimagi/commcare-hq/blob/194506d4bbf2edde0a81a9e6cc66d5752784a7fa/corehq/apps/reports/models.py#L226

Reply:
:( no